pkg/metadata/manager.go

```diff
@@ -28,29 +34,54 @@ type Manager interface {
 	// AllocateArtifactFilePath allocates the file path of the artifact.
 	AllocateArtifactFilePath(src *artifacts.Source, installBinary bool) (string, error)
 
+	// AllocateClusterScopeDirs allocates the directories of one cluster.
+	AllocateClusterScopeDirs() error
```

Comment:
Can we make two `Allocate*` functions as one that the logic is all about metadata setting?

Reply:
done, these two function are now merged into `CreateClusterScopeDirs` methods